src/transformers/models/clip/modeling_clip.py

```diff
@@ -1304,3 +1305,56 @@ def forward(
             hidden_states=vision_outputs.hidden_states,
             attentions=vision_outputs.attentions,
         )
+
+
+@add_start_docstrings(
+    """
+    CLIP Model with a a linear layer on top of the CLIPOutput.image_embeds.
+    """,
+    CLIP_START_DOCSTRING,
+)
+class CLIPForImageClassification(CLIPPreTrainedModel):
+    """
+    Repo with custom implementation: https://github.com/Andron00e/CLIPForImageClassification
+    Pre-trained model on hub: https://huggingface.co/Andron00e/CLIPForImageClassification-v1
+    """
+    def __init__(self, config: CLIPConfig, num_labels: int):
+        super().__init__(config)
+        self.clip = CLIPModel.from_pretrained("openai/clip-vit-base-patch32")
```

Comment:
```suggestion
        self.vision_model = CLIPVisionTransformer(config)
```
In order to make sure people can load a CLIPForImageClassification model with pre-trained CLIP weights, we'll have to use the same variables.